### PR TITLE
Refactor pr workflow

### PR DESCRIPTION
**Pull request by iamer**

### CODE REVIEW

File: .github/workflows/pr-realm-js.yml [hunk 1]
```diff
@@ -0,0 +1,68 @@
...
+        if: ${{ (matrix.variant.os == 'linux') && (steps.cache-prebuild.outputs.cache-hit != 'true') }}
+        run: |
+          sudo apt-get install ninja-build
+
+      # Install node dependencies and build the c++ library
+      - name: Build
+        if: steps.cache-prebuild.outputs.cache-hit != 'true'
+        run: |
+          npm ci --ignore-scripts
+          npm run build-${{ matrix.variant.os }} -- --arch=${{matrix.variant.arch}}
```
Comment (takameyer) [inline .github/workflows/pr-realm-js.yml]: i think i'd like the build script to be of the naming convention `build:{os}`. so for example: 
```
"build:android": ...
"build:linux": ...
etc...
```

File: .github/workflows/pr-realm-js.yml [hunk 2]
```diff
@@ -0,0 +1,68 @@
+name: pr-realm-js
+on: pull_request
+
+jobs:
+  build-prebuild:
+    name: Build for Node.js on ${{ matrix.variant.os }} ${{ matrix.variant.arch }}
```
Comment (takameyer) [inline .github/workflows/pr-realm-js.yml]: Let's remove `Node.js` from the title, as this is actually building for `react-native` and `electron` as well.  We can either add another matrix property, such as {{ matrix.variant.platform }}, or we can just say `Build for ${{ matrix.variant.os}} ${{matrix.variant.arch}}`.
I would go with the latter, unless we find that we are making different builds for different platforms, in which case the former would be more appropriate.

**Review by takameyer — COMMENTED** [review]
Looks like a good start so far.  I only have a couple suggestions.

**Comment by iamer** [comment]
let's continue the discussion in https://github.com/realm/realm-js/pull/4086

**Pull request closed without merging**